Description: Add FrameCreated notifications to RenderViewObserver, so that we can track frames
Author: Chris Coulson <chris.coulson@canonical.com>

diff --git a/content/renderer/render_frame_impl.cc b/content/renderer/render_frame_impl.cc
--- a/content/renderer/render_frame_impl.cc
+++ b/content/renderer/render_frame_impl.cc
@@ -244,16 +244,18 @@ WebKit::WebFrame* RenderFrameImpl::creat
   WebKit::WebFrame* web_frame = WebFrame::create(child_render_frame,
                                                  child_frame_identifier);
 
   if (CommandLine::ForCurrentProcess()->HasSwitch(switches::kSitePerProcess)) {
     g_child_frame_map.Get().insert(
         std::make_pair(web_frame, child_render_frame));
   }
 
+  render_view_->FrameCreated(parent, web_frame);
+
   return web_frame;
 }
 
 void RenderFrameImpl::didDisownOpener(WebKit::WebFrame* frame) {
   render_view_->didDisownOpener(frame);
 }
 
 void RenderFrameImpl::frameDetached(WebKit::WebFrame* frame) {
diff --git a/content/renderer/render_view_impl.cc b/content/renderer/render_view_impl.cc
--- a/content/renderer/render_view_impl.cc
+++ b/content/renderer/render_view_impl.cc
@@ -4508,16 +4508,21 @@ void RenderViewImpl::DidPause(WebKit::We
                                          player->hasAudio(),
                                          false));
 }
 
 void RenderViewImpl::PlayerGone(WebKit::WebMediaPlayer* player) {
   DidPause(player);
 }
 
+void RenderViewImpl::FrameCreated(WebKit::WebFrame* parent,
+                                  WebKit::WebFrame* frame) {
+  FOR_EACH_OBSERVER(RenderViewObserver, observers_, FrameCreated(parent, frame));
+}
+
 void RenderViewImpl::SyncNavigationState() {
   if (!webview())
     return;
 
   const WebHistoryItem& item = webview()->mainFrame()->currentHistoryItem();
   SendUpdateState(item);
 }
 
diff --git a/content/renderer/render_view_impl.h b/content/renderer/render_view_impl.h
--- a/content/renderer/render_view_impl.h
+++ b/content/renderer/render_view_impl.h
@@ -717,16 +717,18 @@ class CONTENT_EXPORT RenderViewImpl
   // appropriate section, add it there. If not, there are some random functions
   // nearer to the top you can add it to.
 
   // Cannot use std::set unfortunately since linked_ptr<> does not support
   // operator<.
   typedef std::vector<linked_ptr<ImageResourceFetcher> >
       ImageResourceFetcherList;
 
+  void FrameCreated(WebKit::WebFrame* parent, WebKit::WebFrame* frame);
+
  protected:
   // RenderWidget overrides:
   virtual void Close() OVERRIDE;
   virtual void OnResize(const ViewMsg_Resize_Params& params) OVERRIDE;
   virtual void DidInitiatePaint() OVERRIDE;
   virtual void DidFlushPaint() OVERRIDE;
   virtual PepperPluginInstanceImpl* GetBitmapForOptimizedPluginPaint(
       const gfx::Rect& paint_bounds,
